homeassistant/components/touchline/climate.py

```diff
@@ -43,6 +49,63 @@ class PresetMode(NamedTuple):
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_HOST): cv.string})
 
 
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Import the configurations from YAML to config flows."""
+    # We only import configs from YAML if it hasn't been imported. If there is a config
+    # entry marked with SOURCE_IMPORT, it means the YAML config has been imported.
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.source == SOURCE_IMPORT:
+            return
+
+    if CONF_HOST in config:
+        host = config[CONF_HOST]
+
+    if not host:
+        _LOGGER.error("No Roth Touchline detected")
+        return
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={CONF_HOST: host},
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Touchline device from config entry."""
+    host = config.data[CONF_HOST]
+    _LOGGER.debug(
+        "Host: %s",
+        host,
+    )
+    py_touchline = PyTouchline()
+    number_of_devices = int(
+        await hass.async_add_executor_job(
+            py_touchline.get_number_of_devices,
+            host,
+        )
+    )
+    _LOGGER.debug(
+        "Number of devices found: %s",
+        number_of_devices,
+    )
+    devices = []
+    for device_id in range(0, number_of_devices):
+        devices.append(Touchline(PyTouchline(device_id)))
+    async_add_entities(devices, True)
```

Comment:
This can be simplified via a comprehension:

```python
async_add_entities(
    Touchline(PyTouchline(device_id)) for device_id in range(0, number_of_devices),
    True,
)
```

Reply:
Doing this, I get this error:
Generator expressions must be parenthesized if not sole argument

Reply:
Ah, right.